src/tuf/trustroot.ts

```diff
@@ -0,0 +1,159 @@
+  // with a story for target discovery.
+  // https://docs.google.com/document/d/1rWHAM2qCUtnjWD4lOrGWE2EIDLoA7eSy4-jB66Wgh0o
+  private async allTargets(): Promise<TargetFile[]> {
+    try {
+      await this.tuf.refresh();
+    } catch (e) {
+      throw new TrustedRootError('error refreshing trust metadata');
+    }
+
+    return Object.values(
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (this.tuf as any).trustedSet.targets?.signed.targets || {}
+    );
+  }
+
+  // Filters the supplied list of targets to those with the specified usage
+  // and returns a new TransparencyLogInstance for each with the associated
+  // public key populated.
+  private async getTLogKeys(
+    targets: TargetFile[],
+    usage: string
+  ): Promise<sigstore.TransparencyLogInstance[]> {
+    const filteredTargets = filterByUsage(targets, usage);
+
+    return Promise.all(
+      filteredTargets.map(async (target) => {
+        const keyBytes = await this.readKey(target);
+        const uri = isTargetMetadata(target.custom.sigstore)
+          ? target.custom.sigstore.uri
+          : '';
+
+        // The log ID is not present in the Sigstore target metadata, but
+        // can be derived by hashing the contents of the public key.
+        return {
+          baseUrl: uri,
+          hashAlgorithm: sigstore.HashAlgorithm.SHA2_256,
+          logId: { keyId: crypto.hash(keyBytes) },
+
+          publicKey: {
+            keyDetails: sigstore.PublicKeyDetails.PKIX_ECDSA_P256_SHA_256,
+            rawBytes: keyBytes,
+          },
+        };
+      })
+    );
+  }
+
+  // Filters the supplied list of targets to those with the specified usage
+  // and returns a new CertificateAuthority populated with all of the associated
+  // certificates.
+  // NOTE: The Sigstore target metadata does NOT provide any mechanism to link
+  // related certificates (e.g. a root and intermediate). As a result, we
+  // assume that all certificates located here are part of the same chain.
+  // This works out OK since our certificate chain verification code tries all
+  // possible permutations of the certificates until it finds one that results
+  // in a valid, trusted chain.
+  private async getCAKeys(
+    targets: TargetFile[],
+    usage: string
+  ): Promise<sigstore.CertificateAuthority[]> {
+    const filteredTargets = filterByUsage(targets, usage);
+
+    const certs = await Promise.all(
+      filteredTargets.map(async (target) => await this.readKey(target))
+    );
+
+    return [
+      {
+        uri: '',
+        subject: undefined,
+        validFor: { start: new Date(0) },
+        certChain: {
+          certificates: certs.map((cert) => ({ rawBytes: cert })),
+        },
+      },
+    ];
+  }
+
+  // Reads the contents of the specified target file, and returns the
+  // contents as a DER-encoded buffer.
+  private async readKey(target: TargetFile): Promise<Buffer> {
```

Comment:
Maybe change the name of this to `pemToDER` or something similar as it's what it's doing. It's not limited to a key.

Reply:
I find myself frustrated that we don't have a single word that encompasses "key or certificate". I updated this function to be named `readTargetBytes` (avoiding any specifics about WHAT I'm reading, but also making it clear that you're getting bytes in the response)